packages/core/src/components/block/block.scss

```diff
@@ -1,28 +1,37 @@
 @import '../../mixins/box-sizing';
+@import '../../../../../typography/utilities/typography-utility';
+
+:host {
+  --primary-block-background-color: var(--color-background-layer-01);
+  --secondary-block-background-color: var(--color-background-layer-02);
+  --text-color: var(--color-foreground-text-strong);
+}
```

Comment:
Host wont work here as component does not <Host syntax in `block.tsx` file. Try replacing `:host` with `.tds-block` as that class is all the time applied.
After it make sure that both Default and Nested story work the same way as compared to prod Storybook. Now it seems that Nested is broken and "Inherit from parent" does not work in any of them.